tests/tests.rs

```diff
@@ -132,3 +133,30 @@ fn test_array_copy (){
     assert_eq!(data.id, 0xbeefedde);
     assert_eq!(name, "hell");
 }
+
+#[derive(Pread, SizeWith)]
+struct Data7A {
+    pub y: u64,
+    pub x: u32,
+}
+
+#[derive(Pread, SizeWith)]
+struct Data7B {
+    pub a: Data7A,
+}
+
+#[test]
+fn test_nested_struct() {
+    const BYTES: [u8; 12] = [
+        // y: u64
+        1, 0, 0, 0, 0, 0, 0, 0,
+        // x: u32
+        2, 0, 0, 0,
+    ];
+    let size = Data7B::size_with(&LE);
+    let mut read = 0;
+    let b: Data7B = BYTES.gread_with(&mut read, LE).unwrap();
+    assert_eq!(b.a.y, 1);
+    assert_eq!(b.a.x, 2);
+    assert_eq!(read, size);
```

Comment:
I think we should also add: `assert_eq!(read, 12)` to check for cases where the value ever changes from our current expectation.

Reply:
Also, I'm curious, what did this return before your PR?

Reply:
`size` was 16 before this patch because the struct presumably has padding.